src/JSONSchema/path-handler.ts

```diff
@@ -0,0 +1,107 @@
+import { JSONSchemaType, JSONSchemaPathInfo } from './types'
+import { useFormContext } from '../components'
+
+const concatFormPath = (path: string, newNode: string): string => {
+  return path + '/' + newNode
+}
+
+const getSplitPath = (path: string): Array<string> => {
+  const split = path.split('/')
+  if (split[0] === '#') {
+    split.shift()
+  }
+  if (split[split.length - 1] === '') {
+    split.pop()
+  }
+  return split
+}
+
+const useObjectFromPath = (path: string): JSONSchemaPathInfo => {
+  const splitPath = getSplitPath(path)
+  let currentOriginal = useFormContext().schema
+  let isRequired = false
+  let objectName = ''
+
+  for (let node = 0; node < splitPath.length; node++) {
+    if (
+      currentOriginal &&
+      currentOriginal.type &&
+      currentOriginal.type === 'object'
+    ) {
+      if (
+        currentOriginal.required &&
+        (currentOriginal.required as Array<string>).indexOf(splitPath[node]) >
+          -1
+      ) {
+        isRequired = true
+      } else {
+        isRequired = false
+      }
+      objectName = splitPath[node]
+      currentOriginal = currentOriginal.properties[splitPath[node]]
+    } else {
+      currentOriginal = {}
+      break
+    }
+  }
+  return [currentOriginal, isRequired, objectName]
+}
+
+const useObjectFromForm = (data: JSONSchemaType): JSONSchemaType => {
+  const originalSchema = useFormContext().schema
+  const orderedSchemaKeys = Object.keys(data).sort()
+  const objectFromData: JSONSchemaType = {}
+
+  for (const key of orderedSchemaKeys) {
+    const splitPath = getSplitPath(key)
+
+    let current = objectFromData
+    let currentOriginal = originalSchema
+
+    if (splitPath) {
+      for (let node = 0; node < splitPath.length; node++) {
+        if (currentOriginal === undefined) {
+          break
+        }
+
+        if (currentOriginal.type && currentOriginal.type === 'object') {
+          currentOriginal = currentOriginal.properties
```

Comment:
I think you are mutating the values from the context (originalSchema). They point to the same object reference.

Create a test with a schema with a deep freezed object to make sure:
https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze